lesson16/src/js/main.js

```diff
@@ -0,0 +1,209 @@
+    categoryTab.setAttribute('role','presentation');
+
+    const categoryButton = document.createElement('button');
+    categoryButton.textContent = target.category;
+    categoryButton.setAttribute('id','tab-' + (i+1));
+    categoryButton.setAttribute('class', 'tab-button js-tabButton');
+    categoryButton.setAttribute('aria-controls','panel-' + (i+1));
+    categoryButton.setAttribute('role', 'tab');
+    categoryButton.setAttribute('aria-selected', 'false');
+    categoryButton.setAttribute('tabindex', '-1');
+    fragment.appendChild(categoryTab).appendChild(categoryButton);
+  });
+  const categoryTabLists = document.querySelector('.tab-list');
+  categoryTabLists.appendChild(fragment);
+}
+
+const getNewsArticle = (data) => {
+  const newsLists = document.createElement('ul');
+  newsLists.classList.add('news-list');
+  const fragment = document.createDocumentFragment();
+  const displayedResults = 5;
+  for (let i = 0; i < displayedResults; ++i){
+    const li = document.createElement('li');
+    li.classList.add('news-item');
+
+    const anchor = document.createElement('a');
+    anchor.classList.add('news-link');
+    anchor.href = data[i].link;
+    anchor.textContent = data[i].title;
+    fragment.appendChild(li).appendChild(anchor);
+    addComment(data[i], li);
+    addNewIcon(data[i],li);
+  }
+  newsLists.appendChild(fragment);
+  return newsLists;
+};
+
+function tabContentsInitialDisplay(data){
+  data.forEach((target) =>{
+    const activeCategory = target.isActive;
+
+    if(activeCategory){
+      const categoryButton = categoryTab.querySelector('.js-tabButton');
+      categoryButton.setAttribute('tabindex', '0');
+      categoryButton.setAttribute('aria-selected','true');
+
+      const getSelectedTadID = categoryButton.getAttribute('aria-controls');
+      const selectedPanel = document.getElementById(getSelectedTadID);
+      selectedPanel.setAttribute('aria-hidden', false);
+    }
+  })
+}
+
+function addNewsImage(data){
+  const getNewsArea = document.querySelectorAll('.news-area');
+
+  data.forEach((category,i) => {
+    const imageArea = document.createElement('div');
+    const img = document.createElement('img');
+    img.src = category.image;
+    img.width = 200;
+    img.height = 200;
+    getNewsArea[i].appendChild(imageArea).appendChild(img);
+  })
+}
+
+function addComment(data, item) {
+  const commentResult = data.comment;
+  if (commentResult.length) {
+    const commentBox = document.createElement('span');
+    const anchor = document.createElement('a');
+    anchor.classList.add('comment-count');
+    anchor.textContent = commentResult.length;
+    item.appendChild(commentBox).appendChild(anchor);
+  }
+}
+
+function addNewIcon(date,item) {
+  const articleDate = date.date;
+  const articleDateToToday = differenceInDays(new Date(),parseISO(articleDate));
+  const judgmenDays = 60;
+  if(articleDateToToday <= judgmenDays){
+    createNewIcon(item)
+  }
+}
+
+function createNewIcon(item){
```

Comment:
IMO
Wouldn't render be more appropriate than create?
```suggestion
function renderNewIcon(item){
```

Reply:
Thanks for the review!
I'll unify the notation between render and create with render throughout the code, since it's shaky!

[fbbe61a](https://github.com/nyattsu72/hanson-frontend-cource/pull/21/commits/fbbe61aa8075606f93890694590d547445ee656c)